storagemarket/helper.go

```diff
@@ -151,13 +151,13 @@ func (c *ChainDealManager) dealIDFromPublishDealsMsg(ctx context.Context, tok ct
 		return dealID, ctypes.EmptyTSK, fmt.Errorf("could not find deal in publish deals message %s", publishCid)
 	}
 
-	if dealIdx >= len(dealIDs) {
```

Comment:
The changes to this helper are required because the return values of IsDealValid have changed.